utils/generate_types/overrides-test.d.ts

```diff
@@ -341,38 +341,92 @@ type AllMatchers<R, T> = PageAssertions & LocatorAssertions & APIResponseAsserti
 
 type IfAny<T, Y, N> = 0 extends (1 & T) ? Y : N;
 type Awaited<T> = T extends PromiseLike<infer U> ? U : T;
-type MakeMatchers<R, T> = {
+type ToUserMatcher<F> = F extends (a: any, ...args: infer Rest) => infer R ? (...args: Rest) => (R extends PromiseLike<infer U> ? Promise<void> : void) : never;
+type ToUserMatcherForArgType<T, ArgType> = T extends (arg: ArgType, ...rest: any[]) => any ? ToUserMatcher<T> : never;
```

Comment:
I'd inline `ToUserMatcher` here:

```suggestion
type ToUserMatcherForArgType<T, ArgType> = T extends (arg: ArgType, ...rest: infer Rest) => infer R ? (...args: Rest) => (R extends PromiseLike<infer U> ? Promise<void> : void) : never;
```